provider/aws/aws.go

```diff
@@ -926,6 +961,39 @@ func (p *AWSProvider) newChange(action route53types.ChangeAction, ep *endpoint.E
 		if useGeolocation {
 			change.ResourceRecordSet.GeoLocation = geolocation
 		}
+
+		geoproximity := &route53types.GeoProximityLocation{}
```

Comment:
what if geoproximity not enabled? why we even have it here?

Reply:
Honestly, I looked at how Geolocation works and tried to follow suit.

If there are no annotations corresponding to GeoProximity, then the `GeoProximityLocation` field of the `ResourceRecordSet` just stays `nil`. The field is populated iff any of the geoproximity annotations are seen.

Also, I wish there was a separate annotation for specifying routing policy. But since that isn't present, I have to check for the presence of all possible geoproximity annotations before I can definitively say that the record is not using geoproximity.

Reply:
This way it could be a breaking change, so unsure